tests/template_tests/test_logging.py

```diff
@@ -75,7 +75,28 @@ def test_log_on_variable_does_not_exist_not_silent(self):
         raised_exception = self.test_handler.log_record.exc_info[1]
         self.assertEqual(
             str(raised_exception),
-            'Failed lookup for key [author] in %r' % ("{%r: %r}" % ('section', 'News'))
+            'Failed lookup for key [author] in %r' % ({'section': 'News'},)
+        )
+
+    def test_log_on_variable_does_not_exist_does_not_stringify(self):
+        class TestObject:
+            def __str__(self):
+                return 'fail'
+
+        test_object = TestObject()
+
+        with self.assertRaises(VariableDoesNotExist):
```

Comment:
If you make this:

```python
with self.assertRaises(VariableDoesNotExist) as exception_ctx:
```
then later on when you need the exception you can just do:

```python
self.assertEqual(
    str(exception_ctx.exception),
   ...
)
```

which looks a bit cleaner than the `exc_info[1]` stuff.